### PR TITLE
API-7908: configuration, bot / agent split

### PR DESCRIPTION
**Pull request by czeczot**

### CODE REVIEW

**Comment by czeczot** [comment]
https://livechatinc.atlassian.net/browse/API-7905

File: src/configuration/structures.ts [hunk 1]
```diff
@@ -63,14 +66,18 @@ export interface WebhookAction {
 }
 
 export interface ListBotsResponse {
-  bot_agents: BotAgent[];
+  bots: Bot[];
```
Comment (janowsiany) [inline src/configuration/structures.ts:69]: Response from ListBots was unnested and now it should return `Bot[]`

File: src/configuration/structures.ts [hunk 2]
```diff
@@ -80,14 +87,12 @@ export enum BotStatus {
 }
 
 export interface GetBotResponse {
-  bot_agent: BotAgentDetails;
+  bot: BotDetails;
```
Comment (janowsiany) [inline src/configuration/structures.ts:90]: Response from GetBot was unnested and now it should return `Bot`

**Review by janowsiany — CHANGES_REQUESTED** [review]

File: src/configuration/structures.ts [hunk 3]
```diff
@@ -79,15 +82,9 @@ export enum BotStatus {
...
-  bot_agent: BotAgentDetails;
-}
-
-export interface BotAgentDetails extends BotAgent {
+export interface BotDetails extends Bot {
   default_group_priority: GroupPriority;
-  application: object;
-  max_chats_count: number;
-  groups: GroupConfig[];
+  client_id: string;
```
Comment (janowsiany) [inline src/configuration/structures.ts:87]: i think it needs to be `owner_client_id`(this field is called like that in response from api), in go you could rename it with json tags but here it is not possible

**Review by janowsiany — CHANGES_REQUESTED** [review]

File: src/configuration/index.ts [hunk 4]
```diff
@@ -112,35 +112,35 @@ export default class ConfigurationAPI extends WebAPI {
...
   }
 
   /**
-   * Lists all bot agents within license.
+   * Lists all bots within license.
    * @param all - switch if should return all bots or only caller's bots
    */
-  async listBots(all?: boolean): Promise<ListBotsResponse> {
-    return this.send("list_bots", { all });
+  async listBots(all?: boolean, fields?: BotFields): Promise<Bot[]> {
```
Comment (kogierman) [inline src/configuration/index.ts:134]: pls describe `fields` param in jsdoc above

File: src/configuration/index.ts [hunk 5]
```diff
@@ -112,35 +112,35 @@ export default class ConfigurationAPI extends WebAPI {
...
 
   /**
-   * Gets info about Bot Agent.
-   * @param bot_agent_id - bot agent ID to get info about
+   * Gets info about Bot.
+   * @param bot_id - bot ID to get info about
    */
-  async getBot(bot_agent_id: string): Promise<GetBotResponse> {
-    return this.send("get_bot", { bot_agent_id });
+  async getBot(bot_id: string, fields?: BotFields): Promise<BotDetails> {
```
Comment (kogierman) [inline src/configuration/index.ts:142]: pls describe `fields` param in jsdoc above

File: src/configuration/structures.ts (hunk 3 above)
Comment (kogierman) [reply src/configuration/structures.ts:87]: ☝️

File: src/configuration/structures.ts [hunk 6]
```diff
@@ -62,15 +65,15 @@ export interface WebhookAction {
...
   id: string;
   name: string;
-  avatar: string;
-  status: BotStatus;
+  avatar_path: string;
+  max_chats_count: number;
+  groups: GroupConfig[];
+  job_title?: string;
+  mobile?: string;
+  work_scheduler?: WorkScheduler;
```
Comment (kogierman) [inline src/configuration/structures.ts:76]: looking at https://github.com/livechat/configuration-api/pull/360/files - all those fields are optional

**Review by kogierman — CHANGES_REQUESTED** [review]
according to https://github.com/livechat/configuration-api/pull/360/files both `list_bots` and `get_bot` will return the same structure, so maybe distinguishing between `Bot` and `BotDetails` is no longer needed?

**Review by janowsiany — APPROVED** [review]

**Review by kogierman — APPROVED** [review]

File: src/configuration/structures.ts [hunk 7]
```diff
@@ -39,14 +39,17 @@ export interface DaySchedule {
 }
 
 export interface CreateBotResponse {
-  bot_agent_id: string;
+  bot_id: string;
```
Comment (janowsiany) [inline src/configuration/structures.ts:42]: `id`

**Review by janowsiany — CHANGES_REQUESTED** [review]

**Review by janowsiany — APPROVED** [review]

**Review by janowsiany — APPROVED** [review]

**Pull request merged**